openwisp_radius/api/freeradius_views.py

```diff
@@ -300,43 +300,99 @@ def get_replies(self, user, organization_id):
-                # avoid crashing on unexpected runtime errors
-                except Exception as e:
-                    logger.exception(f'Got exception "{e}" while executing {counter}')
-                    continue
-                if remaining is None:
-                    continue
-                reply_name = counter.reply_name
-                # send remaining value in RADIUS reply, if needed.
-                # This emulates the implementation of sqlcounter in freeradius
-                # which sends the reply message only if the value is smaller
-                # than what was defined to a previous reply message
-                if reply_name not in data or remaining < self._get_reply_value(
-                    data, counter
-                ):
-                    data[reply_name] = remaining
+            group_checks = get_group_checks(
+                user_group.group, include_additional_checks=True
+            )
+
+            # Validate additional radius checks
+            additional_checks_result = self._validate_additional_radius_checks(
+                group_checks, user, organization_id
+            )
+            if additional_checks_result is not None:
+                return additional_checks_result
+
+            # Execute counter checks
+            counter_result = self._execute_counter_checks(
+                data, user, user_group.group, group_checks
+            )
+            if counter_result is not None:
+                return counter_result
 
         return data, self.accept_status
 
+    def _validate_additional_radius_checks(self, group_checks, user, organization_id):
+        """
+        Validates additional RADIUS checks like Simultaneous-Use
+        Returns rejection response if validation fails, None if validation passes
+        """
+        for check_name in app_settings.ADDITIONAL_RADIUS_CHECKS:
+            group_check = group_checks.pop(check_name, None)
+            if not group_check:
+                continue
+
+            if check_name == "Simultaneous-Use":
+                # Check for simultaneous use
+                if self._check_simultaneous_use(user, group_check):
+                    data = self.reject_attributes.copy()
+                    if "Reply-Message" not in data:
+                        data["Reply-Message"] = (
+                            "You are already logged in - access denied"
+                        )
+                    return data, self.reject_status
+        return None
+
+    def _check_simultaneous_use(self, user, group_check):
+        """
+        Check if user has exceeded simultaneous use limit
+        Returns True if limit is exceeded, False otherwise
+        """
+        max_simultaneous = int(group_check.value)
+        if max_simultaneous <= 0:
+            return False
+        # Count open sessions for this user
+        open_sessions = RadiusAccounting.objects.filter(
+            username=user.username, stop_time__isnull=True
```

Comment:
What if the same user has 2 radius accountings in two different organizations with 2 different devices?

Quite unlikely, but the goal is to support multitenancy, right? So I believe we should also add the org here.